From 19a29b523eef31b5d6c08abed553c27cd33c07f0 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?G=C3=A1bor=20Gomb=C3=A1s?= <gombasg@sztaki.hu>
Date: Tue, 8 Jan 2008 20:52:31 +0100
Subject: [PATCH] Fix "reconstruct -r"

The comment is wrong: mboxlist_findall() does not seem to remove the domain
part therefore the pattern is invalid. Remove the domain by hand to make the
pattern correct.
---
 imap/reconstruct.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/imap/reconstruct.c b/imap/reconstruct.c
index 526cfc2..a89bd1d 100644
--- a/imap/reconstruct.c
+++ b/imap/reconstruct.c
@@ -339,7 +339,8 @@ int main(int argc, char **argv)
 					    fflag ? &head : NULL);
 	if (rflag) {
 	    /* build a pattern for submailboxes */
-	    /* XXX mboxlist_findall() is destructive and removes domain */
+	    char *p = strchr(buf, '@');
+	    if (p) *p = '\0';
 	    strlcat(buf, ".*", sizeof(buf));
 
 	    /* append the domain */
